packages/transformers/js/core/src/collect.rs

```diff
@@ -241,6 +245,15 @@ impl Visit for Collect {
     }
     self.in_module_this = false;
 
+    for (_key, node) in self.this_exprs.clone() {
+      if let MemberProp::Ident(prop) = &node.prop {
```

Comment:
There are two identical `if let MemberProp::Ident(prop)` checks, one here and one below where `self.this_expr` is populated.

Reply:
need both `if let MemberProp::Ident(prop)` because build fails without casting to a MemberProp::Ident